fmn/backends/base.py

```diff
@@ -53,7 +53,6 @@ def base_url_with_trailing_slash(self) -> str:
     def api_url(self) -> str | None:
         return self.base_url
 
-    @ft_cache
     def __str__(self) -> str:
```

Comment:
This must be the potential memory leak 😉, only it shouldn’t be…

The reason why I cached `APIClient.__str__()` was to speed up cache key generation for `cashews` and I’d like to retain that optimization of possible (it speeds up `str()` by a factor of about 3). And I think it is, we could do either or both of:

* Make the proxy objects singletons based off the arguments passed to the class when constructing (and `Requester` as well while we’re at it) ⇒ Only one return value cached per singleton/class.
* Clear the cache on the method when finalizing in `__del__()`.

Reply:
Or we could store the computed value in a local class variable such as `_str` and just return that when `__str__()` is called. I'd prefer this or the cache clear on `__del__()`, any preference between the two?

(I kinda dislike singletons as they can make testing harder)

Reply:
Good idea, something like this should do the trick:

```
def __str__(self) -> str:
    try:
        return self._str
    except AttributeError:
        clsname = type(self).__name__
        self._str = f"{clsname}({self.base_url!r})"
        return self._str
```